src/envoy/auth/config.cc

```diff
@@ -15,199 +15,119 @@
-      Http::Headers::get().MethodValues.Get);
-  message->headers().insertPath().value(path);
-  message->headers().insertHost().value(host);
-
-  request_ = cm_.httpAsyncClientForCluster(cluster_->name())
-                 .send(std::move(message), *this, timeout_);
+Config::Config(std::vector<IssuerInfo>&& issuers)
+    : issuers_(std::move(issuers)) {
+  ENVOY_LOG(debug, "Config: {}", __func__);
 }
 
-void AsyncClientCallbacks::Cancel() { request_->cancel(); }
+Config::Config(const Json::Object& config) {
+  ENVOY_LOG(debug, "Config: {}", __func__);
 
-IssuerInfo::IssuerInfo(Json::Object *json) {
-  ENVOY_LOG(debug, "IssuerInfo: {}", __func__);
-  // Check "name"
-  name_ = json->getString("name", "");
-  if (name_ == "") {
-    ENVOY_LOG(debug, "IssuerInfo: Issuer name missing");
-    failed_ = true;
+  // Load the issuers as JSON array.
+  std::vector<Json::ObjectSharedPtr> issuer_jsons;
+  try {
+    issuer_jsons = config.getObjectArray("issuers");
+  } catch (...) {
+    ENVOY_LOG(error, "Config: issuers should be array type");
     return;
   }
-  // Check "audience". It will be an empty array if the key "audience" does not
-  // exist
+
+  for (auto issuer_json : issuer_jsons) {
+    IssuerInfo issuer;
+    if (LoadIssuerInfo(*issuer_json, &issuer)) {
+      std::string err = issuer.Validate();
+      if (err.empty()) {
+        issuers_.push_back(issuer);
+      } else {
+        ENVOY_LOG(error, "Issuer [name = {}]: {}", issuer.name, err);
+      }
+    }
+  }
+}
+
+bool Config::LoadIssuerInfo(const Json::Object& json, IssuerInfo* issuer) {
+  // Check "name"
+  issuer->name = json.getString("name", "");
+  if (issuer->name == "") {
+    ENVOY_LOG(error, "Issuer: Issuer name missing");
+    return false;
+  }
+
+  // Check "audience".
+  // It will be an empty array if the key "audience" does not exist.
   try {
-    audiences_ = json->getStringArray("audiences", true);
+    auto audiences = json.getStringArray("audiences", true);
+    issuer->audiences.insert(audiences.begin(), audiences.end());
   } catch (...) {
-    ENVOY_LOG(debug, "IssuerInfo [name = {}]: Bad audiences", name_);
-    failed_ = true;
-    return;
+    ENVOY_LOG(error, "Issuer [name = {}]: Bad audiences", issuer->name);
+    return false;
   }
+
   // Check "pubkey"
   Json::ObjectSharedPtr json_pubkey;
   try {
-    json_pubkey = json->getObject("pubkey");
+    json_pubkey = json.getObject("pubkey");
   } catch (...) {
-    ENVOY_LOG(debug, "IssuerInfo [name = {}]: Public key missing", name_);
-    failed_ = true;
-    return;
+    ENVOY_LOG(error, "Issuer [name = {}]: Public key missing", issuer->name);
+    return false;
   }
+
   // Check "type"
-  std::string pkey_type_str = json_pubkey->getString("type", "");
-  if (pkey_type_str == "pem") {
-    pkey_type_ = Pubkeys::PEM;
-  } else if (pkey_type_str == "jwks") {
-    pkey_type_ = Pubkeys::JWKS;
+  std::string pubkey_type_str = json_pubkey->getString("type", "");
```

Comment:
Should this be wrapped with try/catch. 

Alternatively, you can wrap the whole function in a big try block, catch exception and print out its message as above. It's probably easier than check with json function throw exception or not.